src/editor.rs

```diff
@@ -835,16 +840,22 @@ enum PromptState {
     Cancelled,
 }
 
+thread_local! (static CHARACTER: RefCell<Vec<u8>> = {let cache = Vec::new(); RefCell::new(cache)});
 /// Process a prompt keypress event and return the new state for the prompt.
 fn process_prompt_keypress(mut buffer: String, key: &Key) -> PromptState {
     match key {
         Key::Char(b'\r') => return PromptState::Completed(buffer),
         Key::Escape | Key::Char(EXIT) => return PromptState::Cancelled,
         Key::Char(BACKSPACE | DELETE_BIS) => _ = buffer.pop(),
         Key::Char(c @ 0..=126) if !c.is_ascii_control() => buffer.push(*c as char),
+        Key::Char(c @ 128..=255) => CHARACTER.with(|cache| cache.borrow_mut().push(*c)),
         // No-op
         _ => (),
     }
+    let character = CHARACTER.with(|cache| String::from_utf8(cache.borrow_mut().clone()));
+    let _ = character.clone().map_or((), |c| buffer.push_str(c.as_str()));
+    let _ = character.map_or((), |_| CHARACTER.with(|cache| cache.borrow_mut().clear()));
```

Comment:
I think this would be clearer as a single `if` instead of 3 assignments, something like

```rust
if let Ok(s) = std::str::from_utf8(&cache) {
  buffer.push_str(s);
  cache.clear();
}
```

What do you think?

Note that using `std::str::from_utf8` instead of `String::from_utf8` avoids allocating an extra string, and there is no need for mutable borrow + cloning, a simple borrow is enough